gui/src/app/compileStanProgram/mainJsUrlCache.ts

```diff
@@ -0,0 +1,69 @@
+export const checkMainJsUrlCache = async (
+  stanProgram: string,
+): Promise<string | null> => {
+  const mainJsCache = localStorage.getItem("mainJsCache");
+  if (!mainJsCache) return null;
+  try {
+    const cache = JSON.parse(mainJsCache);
```

Comment:
Is there a solution that doesn't involve parsing/writing to JSON so frequently? I've read a bit about react's `useSyncExternalStore` hook which seems like it might be for this kind of thing?

Reply:
We can consider it I suppose. But I want to emphasize that this is going to be a rather small string. A lot more expensive things are happening each time user saves a file.